src/detector.c

```diff
@@ -1642,110 +1652,255 @@ void test_detector(char *datacfg, char *cfgfile, char *weightfile, char *filenam
 
-            FILE* fw = fopen(labelpath, "wb");
-            int i;
-            for (i = 0; i < nboxes; ++i) {
-                char buff[1024];
-                int class_id = -1;
-                float prob = 0;
-                for (j = 0; j < l.classes; ++j) {
-                    if (dets[i].prob[j] > thresh && dets[i].prob[j] > prob) {
-                        prob = dets[i].prob[j];
-                        class_id = j;
+            // pseudo labeling concept - fast.ai
+            if (save_labels)
+            {
+                char labelpath[4096];
+                replace_image_to_label(input, labelpath);
+
+                FILE* fw = fopen(labelpath, "wb");
+                int i;
+                for (i = 0; i < nboxes; ++i) {
+                    char buff[1024];
+                    int class_id = -1;
+                    float prob = 0;
+                    for (j = 0; j < l.classes; ++j) {
+                        if (dets[i].prob[j] > thresh && dets[i].prob[j] > prob) {
+                            prob = dets[i].prob[j];
+                            class_id = j;
+                        }
+                    }
+                    if (class_id >= 0) {
+                        sprintf(buff, "%d %2.4f %2.4f %2.4f %2.4f\n", class_id, dets[i].bbox.x, dets[i].bbox.y, dets[i].bbox.w, dets[i].bbox.h);
+                        fwrite(buff, sizeof(char), strlen(buff), fw);
                     }
                 }
-                if (class_id >= 0) {
-                    sprintf(buff, "%d %2.4f %2.4f %2.4f %2.4f\n", class_id, dets[i].bbox.x, dets[i].bbox.y, dets[i].bbox.w, dets[i].bbox.h);
-                    fwrite(buff, sizeof(char), strlen(buff), fw);
-                }
+                fclose(fw);
             }
-            fclose(fw);
+
+            free_detections(dets, nboxes);
+            free_image(im);
+            free_image(sized);
+
+            if (!dont_show) {
+                wait_until_press_key_cv();
+                destroy_all_windows_cv();
+            }
+
+            if (filename) break;
         }
 
-        free_detections(dets, nboxes);
-        free_image(im);
-        free_image(sized);
+        if (json_file) {
+            char *tmp = "\n]";
+            fwrite(tmp, sizeof(char), strlen(tmp), json_file);
+            fclose(json_file);
+        }
 
-        if (!dont_show) {
-            wait_until_press_key_cv();
-            destroy_all_windows_cv();
+        // free memory
+        free_ptrs((void**)names, net.layers[net.n - 1].classes);
+        free_list_contents_kvp(options);
+        free_list(options);
+
+        int i;
+        const int nsize = 8;
+        for (j = 0; j < nsize; ++j) {
+            for (i = 32; i < 127; ++i) {
+                free_image(alphabet[j][i]);
+            }
+            free(alphabet[j]);
         }
+        free(alphabet);
 
-        if (filename) break;
+        free_network(net);
     }
+    else{
+        DIR *folder;
+        struct dirent *entry;
```

Comment:
these two lines, for example, depend on dirent.h being included, which is incompatible with our aim of being 100% windows compatible

Reply:
okay, I noted it, I'll apply required changes

Reply:
what's your opinion for [Boost filesystem](https://www.boost.org/doc/libs/1_43_0/libs/filesystem/doc/index.htm)? I don't have much experience in building projects in windows. all your help is appreciated

Reply:
if you can use std::filesystem it would be much better, so that we can avoid depending on another external lib

Reply:
Okay, I'll look into it
thanks

Reply:
Hi, @cenit I was planning to fix this PR, I just wanted to know if there are any other changes to make other than filesystem?